src/waitress/channel.py

```diff
@@ -44,10 +44,13 @@ class HTTPChannel(wasyncore.dispatcher):
     last_activity = 0  # Time of last activity
     will_close = False  # set to True to close the socket.
     close_when_flushed = False  # set to True to close the socket when flushed
-    requests = ()  # currently pending requests
     sent_continue = False  # used as a latch after sending 100 continue
     total_outbufs_len = 0  # total bytes ready to send
     current_outbuf_count = 0  # total bytes written to current outbuf
+    client_disconnected = False
```

Comment:
how is this variable different from `self.connected` which is used to raise `ClientDisconnected` exceptions?

Reply:
In general I'm concerned that the code hasn't been updated with any real coordination around `close_when_flushed`, `client_disconnected`, `connected`, `will_close`, etc. I'd like a sense for when you think each one is currently being used to justify adding a new variable.

Reply:
Thanks, I will have to think a bit more about these. When starting with this, I was hoping not to have to deal with the response end of the request processing, but I guess that was too optimistic. Is there any documentation that could help me besides reading the source code and maybe asking questions?

Reply:
Hm if I use `not self.connected` instead of `self.client_disconnected`, the tests still all succeed. But from what I can see in the code, it seems to me that this might prevent some cleanup (like the calls to `request.close()` in `service()` and the flushing of the output buffers in `handle_close()`).
@mmerickel or @bertjwregeer, can you confirm or alleviate my suspicion?